shared/wallets/transaction/index.js

```diff
@@ -0,0 +1,180 @@
+      */
+      break
+  }
+
+  const amount = <Text type={textTypeSemibold}>{props.amountUser}</Text>
+
+  if (props.counterpartyType === 'wallet') {
+    if (props.yourRole === 'sender') {
+      return (
+        <Text type={textType}>
+          You transferred Lumens worth {amount} from this wallet to {counterparty}.
+        </Text>
+      )
+    }
+
+    return (
+      <Text type={textType}>
+        You transferred Lumens worth {amount} from {counterparty} to this wallet.
+      </Text>
+    )
+  }
+
+  if (props.yourRole === 'sender') {
+    return (
+      <Text type={textType}>
+        You sent Lumens worth {amount} to {counterparty}.
+      </Text>
+    )
+  }
+
+  return (
+    <Text type={textType}>
+      {counterparty} sent you Lumens worth {amount}.
+    </Text>
+  )
+}
+
+type AmountXLMProps = {|
+  yourRole: Role,
+  amountXLM: string,
+  pending: boolean,
+|}
+
+const AmountXLM = (props: AmountXLMProps) => {
+  const color = props.pending
+    ? globalColors.black_20
+    : props.yourRole === 'sender' ? globalColors.red : globalColors.green
+  const amount = `${props.yourRole === 'sender' ? '-' : '+'} ${props.amountXLM}`
+    // Replace spaces with non-breaking spaces.
+    .replace(/ /g, '\u00a0')
+  return (
+    <Text style={{color}} type="BodyExtrabold">
+      {amount}
+    </Text>
+  )
+}
+
+export type Props = {|
+  ...$Exact<DetailProps>,
+  ...$Exact<IconProps>,
+  // A null timestamp means the transaction is still pending.
+  timestamp: Date | null,
+  amountXLM: string,
+  note: string,
+|}
+
+export const Transaction = (props: Props) => (
+  <Box2 direction="horizontal" fullWidth={true} style={styles.container}>
+    <Icon counterparty={props.counterparty} counterpartyType={props.counterpartyType} large={props.large} />
+    <Box2 direction="vertical" fullHeight={true} fullWidth={true} style={styles.rightContainer}>
+      <Text type="BodySmall">{props.timestamp ? formatTimeForPopup(props.timestamp) : 'Pending'}</Text>
+      <Box2 direction="horizontal" fullHeight={true} fullWidth={true} style={styles.rightDownContainer}>
+        <Box2 direction="vertical" fullHeight={true} fullWidth={true} style={styles.detailContainer}>
+          <Detail
+            large={props.large}
+            yourRole={props.yourRole}
+            counterparty={props.counterparty}
+            counterpartyType={props.counterpartyType}
+            amountUser={props.amountUser}
+          />
+          {props.large && (
+            <Text style={styles.note} type="Body">
+              {props.note}
+            </Text>
+          )}
+        </Box2>
+        <AmountXLM pending={!props.timestamp} yourRole={props.yourRole} amountXLM={props.amountXLM} />
+      </Box2>
+    </Box2>
+  </Box2>
+)
+
+const styles = styleSheetCreate({
+  container: {
+    padding: globalMargins.tiny,
+    paddingRight: globalMargins.small,
+  },
+  detailContainer: {},
+  note: platformStyles({
+    // TODO: Consider using markdown quoting.
```

Comment:
The story mentioned this, but I'm a bit wary of running the whole markdown parser over the note...

Reply:
why? the notes are small and we run markdown on every single piece of chat ever

Reply:
A few reasons:

1) The notes are a stellar thing that's not inherently markdown, right? Seems wrong to assume they are.
2) Just because it's safe in chat doesn't mean it's safe in notes from rando Stellar addresses. For example, I don't think we want to linkify anything in a note, do we?
3) Our current method of quoting doesn't always work given ``` blocks, etc.